src/structure.pyx

```diff
@@ -416,6 +419,102 @@ cdef class Structure:
             freesasa_structure_free(self._c_structure)
 
 
+def getStructures(fileName,
+                   options = Structure.defaultStructureArrayOptions,
+                   classifier = None, *,
+                   selection = None):
```

Comment:
I'm not sure, but would it make more sense to make `selection` a part of the options object and merge this with `structureArray()`, and calling the option `chain-groups`,  so it's more similar to the CLI? (Perhaps keeping this function privately and calling it if necessary from `structureArray`)

Reply:
That wouldn't be a breaking change, just an added feature.

Reply:
Right, I thought that would be indeed a breaking change because:

`Structure._get_structure_options()` turns that dict into a bitfield that, if I'm not wrong, maps onto the C API's  `freesasa_structure_options` [enum](https://github.com/mittinatten/freesasa/blob/aa3c67d174049c0103d39a2a97b4a3a931d55243/src/freesasa.h#L182).

So even if we added a `FREESASA_SEPARATE_CHAINS` to the enum, we would still need an extra argument for the selection string. The C version of `getStructures()` solves this by getting the CLI `state`.

Did I get it right?

Reply:
I had a computer breakdown, so it’ll be a couple more days before I can have proper look at this.

I haven’t looked at the code that converts to the option bit field in a while, but it doesn’t necessarily have to be a problem that that ignores some options in the dict. There’s no reason the Python interface has to have an identical structure to the C interface, and we don’t need to add it to the enum. If it’s more convenient to merge the two sets of options into one dict, I don’t think it has to be a problem that under the hood different parts are handled differently, since the conversion function is private. There might be some other consequences though, that I don’t see?

Reply:
Ugh, they always break at the worst time possible. Anyway, no hurries.

So, `_get_structure_options()` would return `Tuple[int, Optional[str]]`, with the selection string (eg: `AB+C`) being the second optional argument. Then I would replace every call to it from something like:
```
optbitfield = Structure._get_structure_options(options)
```
to:
```
optbitfield, _ = Structure._get_structure_options(options)
```
Except for the call from `getStructures()`, which would be the only one to actually use the selection string.

Reply:
I  was thinking simply let `chain-groups` be ignored in `Structure._get_structure_options` (but it has to be added to `knownOptions`. Perhaps the code that checks for known options could be extracted to a static method `Strcture._validate_options(options)`, to indicate that what are valid options is not determined by the bitfield-conversion-function. `_get_structure_options` could also  get a name that reflects this change,  e.g. `_get_bitfield_from_options`. Since these are private, some renaming and refactoring should be ok.

Reply:
Added `Structure._get_bitfield_from_options()`, `Structure._get_structure_options()` and replaced `Structure._get_structure_options()` by them.